src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -192,7 +193,31 @@ ThreadError DatasetManager::Register(void)
 
     VerifyOrExit((message = mSocket.NewMessage(0)) != NULL, error = kThreadError_NoBufs);
     SuccessOrExit(error = message->Append(header.GetBytes(), header.GetLength()));
-    SuccessOrExit(error = message->Append(mLocal.GetBytes(), mLocal.GetSize()));
+
+#if OPENTHREAD_ENABLE_COMMISSIONER
+    isCommissioner = mNetif.GetCommissioner().GetState() > MeshCoP::Commissioner::kStateDisabled ? true : false;
+#endif  // OPENTHREAD_ENABLE_COMMISSIONER
+
+    if (!isCommissioner)
+    {
+        SuccessOrExit(error = message->Append(mLocal.GetBytes(), mLocal.GetSize()));
+    }
+    else
+    {
+        const Tlv *cur = reinterpret_cast<const Tlv *>(mLocal.GetBytes());
+        const Tlv *end = reinterpret_cast<const Tlv *>(mLocal.GetBytes() + mLocal.GetSize());
+
+        while (cur < end)
+        {
+            Tlv::Type type = cur->GetType();
```

Comment:
Not sure the reason that the comments are not shown in github.

If not excluding the TLVs that affect connectivity, how the native commissioner send the latest active dataset to Leader?